### PR TITLE
Add github action to automatically push to release branch on release

### PR DESCRIPTION
**Pull request by Hedingber**
In order for the action to work a secret should be set in the repo named `RELEASE_GITHUB_ACCESS_TOKEN` and the value should be an access token with workflow and repo scopes.
I'll set it in the `mlrun/demos` with an access token of @mlrun-iguazio

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    name: Push source to release branch
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Push to release branch
+      run: |
+        export tag=$(echo ${GITHUB_REF#refs/tags/})
+        export tag_match_pattern=$(echo "$tag" | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$")
+        export tag_suffix=$(echo ${tag:0:$((${#tag}-1))}x)
```
Comment (omesser) [inline .github/workflows/release.yaml:18]: vX.Y.Z
Z (as well as x,y,z) could be multiple digits - if I understand the syntax correctly you only remove the last char

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        export branch_name=$(echo release/$tag_suffix)
+        if [ -z "$tag_match_pattern" ]; \
```
Comment (omesser) [inline .github/workflows/release.yaml:20]: https://stackoverflow.com/questions/58862864/github-actions-ci-conditional-regex

**Review by omesser — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (Hedingber) [reply .github/workflows/release.yaml]: Nice catch, fixed

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        export branch_name=$(echo release/$tag_suffix)
+        git fetch origin $branch_name || exit_code=$?
+        if [[ $exit_code -eq 0 ]]; \
+          then \
+            echo "Branch $branch_name exists. Adding changes"; \
+            git checkout $branch_name; \
+            git merge $tag; \
+          else \
+            echo "Creating new branch: $branch_name"; \
+            git checkout -b $branch_name; \
```
Comment (omesser) [inline .github/workflows/release.yaml:35]: 👏

**Review by omesser — APPROVED** [review]

**Pull request merged**